cobalt/dom/source_buffer_metrics.cc

```diff
@@ -63,44 +63,89 @@ double GetWallToThreadTimeRatio(int64_t wall_time, int64_t thread_time) {
 }  // namespace
 SB_ONCE_INITIALIZE_FUNCTION(StatisticsWrapper, StatisticsWrapper::GetInstance);
 
+#endif  // !defined(COBALT_BUILD_TYPE_GOLD)
+
 void SourceBufferMetrics::StartTracking() {
   if (!is_primary_video_) {
     return;
   }
 
   DCHECK(!is_tracking_);
   is_tracking_ = true;
-  wall_start_time_ = starboard::CurrentMonotonicTime();
+  wall_start_time_ = clock_->NowTicks();
   thread_start_time_ = starboard::CurrentMonotonicThreadTime();
 }
 
-void SourceBufferMetrics::EndTracking(std::size_t size_appended) {
+void SourceBufferMetrics::EndTracking(SourceBufferMetricsAction action,
+                                      std::size_t size_appended) {
   if (!is_primary_video_) {
     return;
   }
 
   DCHECK(is_tracking_);
   is_tracking_ = false;
 
-  int64_t wall_duration = starboard::CurrentMonotonicTime() - wall_start_time_;
+  base::TimeDelta wall_duration = clock_->NowTicks() - wall_start_time_;
   int64_t thread_duration =
       starboard::CurrentMonotonicThreadTime() - thread_start_time_;
-  total_wall_time_ += wall_duration;
+
+  RecordTelemetry(action, wall_duration);
+
+#if !defined(COBALT_BUILD_TYPE_GOLD)
+  total_wall_time_ += wall_duration.InMicroseconds();
   total_thread_time_ += thread_duration;
 
   total_size_ += size_appended;
 
   if (size_appended > 0) {
-    int wall_bandwidth = GetBandwidth(size_appended, wall_duration);
+    int wall_bandwidth =
+        GetBandwidth(size_appended, wall_duration.InMicroseconds());
     int thread_bandwidth = GetBandwidth(size_appended, thread_duration);
 
     max_wall_bandwidth_ = std::max(max_wall_bandwidth_, wall_bandwidth);
     min_wall_bandwidth_ = std::min(min_wall_bandwidth_, wall_bandwidth);
     max_thread_bandwidth_ = std::max(max_thread_bandwidth_, thread_bandwidth);
     min_thread_bandwidth_ = std::min(min_thread_bandwidth_, thread_bandwidth);
   }
+#endif  // !defined(COBALT_BUILD_TYPE_GOLD)
+}
+
+void SourceBufferMetrics::RecordTelemetry(
+    SourceBufferMetricsAction action, const base::TimeDelta& action_duration) {
+  switch (action) {
+    case SourceBufferMetricsAction::PREPARE_APPEND:
+      UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
+          "Cobalt.Media.SourceBuffer.PrepareAppend.Timing", action_duration,
+          base::TimeDelta::FromMicroseconds(1), base::TimeDelta::FromSeconds(1),
+          50);
```

Comment:
Were you able to test these histograms locally? How was your bucket distribution? I wonder about your min/max ranges here and your bucket size (50). 1 microsecond to 1 second is a delta of 999,999. This is a very wide range... is 50 buckets too coarse? Can you tighten your interval here reasonably and also perhaps up the buckets to 100?

Reply:
Looks good.